initial implementation of mqtt executor

SimonT90poe opened the pull request:
Add new executor to support MQTT message brokers

Signed-off-by: Simon Truss <Simon.Truss@90poe.io>

ovh-cds commented:
CDS Report build-venom-a#597.0 ■   
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ■

ovh-cds commented:
CDS Report build-venom-a#597.0 ■   
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ■

ovh-cds commented:
CDS Report build-venom-a#606.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

ovh-cds commented:
CDS Report build-venom-a#609.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

yesnault commented:
@SimonT90poe  thank you for this new executor.

Tests are failing, output:

```
 • MQTT testsuite (./mqtt.yml)
 	• MQTT-publish-testcase SUCCESS
	  [trac] writing MQTT-testsuite.MQTT-publish-testcase.step.0.dump.json
 	• MQTT-subscribe-testcase FAILURE
	  [trac] writing MQTT-testsuite.MQTT-subscribe-testcase.step.0.dump.json
	  [trac] writing MQTT-testsuite.MQTT-subscribe-testcase.step.1.dump.json
	  [trac] writing MQTT-testsuite.MQTT-subscribe-testcase.step.2.dump.json
Testcase "MQTT subscribe testcase", step #2: Assertion "result.messages.__len__ ShouldEqual 1" failed. expected: 1  got: 0 (./mqtt.yml:49)
Testcase "MQTT subscribe testcase", step #2: Assertion "result.messagesjson.__len__ ShouldEqual 1" failed. expected: 1  got: 0 (./mqtt.yml:50)
Testcase "MQTT subscribe testcase", step #2: Assertion "result.messagesjson.messagesjson0.c ShouldEqual 1" failed. expected: 1  got: <nil> (./mqtt.yml:51)
Testcase "MQTT subscribe testcase", step #2: Assertion "result.topics.topics0 ShouldEqual \"venom/mqtt_test/topic\"" failed. expected: venom/mqtt_test/topic  got: <nil> (./mqtt.yml:0)
```
Can you take a look at this please ?

yesnault commented on executors/mqtt/mqtt.go:
```diff
@@ -0,0 +1,323 @@
+	ClientID            string `json:"client_id" yaml:"clientId"`
+
+	// Subscription topic
+	Topics []string `json:"topic" yaml:"topic"`
```
shouldn't be topics in annotation?

SimonT90poe replied:
yes, good catch.

SimonT90poe replied:
resolved in 9e989b9

SimonT90poe commented:
> @SimonT90poe thank you for this new executor.
> 
> Tests are failing, output:
[...]
> 
> Can you take a look at this please ?

The v2 docker container had changed the default behaviour and removed the unsecured listener. Added a config file to support MQTT over TCP.

resolved in 3488d70

ovh-cds commented:
CDS Report build-venom-a#628.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

ovh-cds commented:
CDS Report build-venom-a#629.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

ovh-cds commented:
CDS Report build-venom-a#630.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

SimonT90poe commented:
Where do I find the build logs, because this was working on my own machine. The only possible issue I can think of is the dependency of the docker container I added.

ovh-cds commented:
CDS Report build-venom-a#631.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

ovh-cds commented:
CDS Report build-venom-a#632.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

ovh-cds commented:
CDS Report build-venom-a#633.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

ovh-cds commented:
CDS Report build-venom-a#634.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

ovh-cds commented:
CDS Report build-venom-a#635.0 ✘  
* Build
  * Build ✔ 
* Tests
  * Acceptance Tests ✘

yesnault commented:
@SimonT90poe  Sorry for notification on this PR. It's ok now:

```
[INFO]  • MQTT testsuite (./mqtt.yml)
[INFO]  	• MQTT-publish-testcase SUCCESS
[INFO] 	  [trac] writing MQTT-testsuite.MQTT-publish-testcase.step.0.dump.json
[INFO]  	• MQTT-subscribe-testcase SUCCESS
[INFO] 	  [trac] writing MQTT-testsuite.MQTT-subscribe-testcase.step.0.dump.json
[INFO] 	  [trac] writing MQTT-testsuite.MQTT-subscribe-testcase.step.1.dump.json
[INFO] 	  [trac] writing MQTT-testsuite.MQTT-subscribe-testcase.step.2.dump.json
[INFO] 	  [trac] writing MQTT-testsuite.MQTT-subscribe-testcase.step.3.dump.json
```
